### ISSUE TITLE
Are there any plans to make nmslib installable on Python 3.12?

### ISSUE THREAD

**Issue by Slasher111** [nmslib/nmslib#555]
Actually, the whole issue is described in the title :). I suppose that a lot of peope are waiting for your brilliant library to work in the modern Python versions. Thank you in advance!

**Comment by mfide** [comment]
+1

**Comment by searchivarius** [comment]
Yes, there are plans, but I need a vacation to finish this. The problem is that I need to transition to github actions. Any help with this is highly appreciated. Travis is allegedly supporting OSS, but previously builds worked, but now they fail with random errors. Appveyor also stopped working (this is for Windows). So, I guess it will be just easier to port everything to GitHub actions. This was the only real blocker to update the library.

**Comment by phw** [comment]
@searchivarius there is the temporary fork https://github.com/metabrainz/nmslib-metabrainz/, including Python 3.11/3.12 support and building with GitHub actions.

The commits are somewhat all over the place, especially for GitHub actions, as it was tried to get things going. But I think this would be turned into clean pull requests.

**Comment by searchivarius** [comment]
Fantastic, many thanks. Let me try to incorporate this!

**Comment by AlbertoMQ** [comment]
Did this ever get added?

**Comment by searchivarius** [comment]
I realized, I will just build these manually for now. I had to fix some Windows-related issues, so this delayed me a bit. But I think I can build all the wheels without github actions (at least for now).

**Comment by celsofranssa** [comment]
Any update on this?

**Comment by searchivarius** [comment]
Hi folks, 
We now support all "live" Python versions: 3.8 through 3.13. I will add 3.14 when it becomes more mainstream (currently it's a pre-release version).
As usual, building from scratch is a bit tricky, but I hope precompiled binaries are solving the problem. Please, let me know if there are any issues.

**Comment by searchivarius** [comment]
@phw feel free to switch back to using this main branch. We now have a comprehensive build and test script that covers three OS (Windows, MAC, and Linux) and two CPUs (six combinations in total).
